### PR TITLE
Add world.BlockSource interface to allow external libraries to use DF block models

### PR DESCRIPTION
**Pull request by ethaniccc**

### CODE REVIEW

**Review by Sandertv — COMMENTED** [review]
Hi, thanks for the PR! I think this is a good idea, but I would consider moving `world.BlockSource` to `model.BlockSource`, as that is where the interface is used. Otherwise looks good.

File: server/world/block_source.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import "github.com/df-mc/dragonfly/server/block/cube"
+
+// BlockSource represents a source for blocks. Blocks can be retrieved
+// and set in the block source.
+type BlockSource interface {
+	// Block returns the block at the given position in the block source.
+	Block(cube.Pos) Block
+	// SetBlock sets the block at the given position in the block source.
+	SetBlock(cube.Pos, Block, *SetOpts)
```
Comment (Sandertv) [inline server/world/block_source.go:11]: I believe SetBlock isn't used anywhere in this package, so I think you could remove this method from the interface.

**Comment by ethaniccc** [comment]
> Hi, thanks for the PR! I think this is a good idea, but I would consider moving `world.BlockSource` to `model.BlockSource`, as that is where the interface is used. Otherwise looks good.

That won't be possible without moving `world.BlockModel`, as it would cause an import cycle with `model` <-> `world`

**Review by Sandertv — APPROVED** [review]
Looks good, sensible change! Thank you for the PR!

**Pull request merged**